.github/workflows/ci-workflow.yml

```diff
@@ -38,26 +51,26 @@ jobs:
 
     steps:
       - name: Clone
-        uses: actions/checkout@v2
+        uses: actions/checkout@v4
 
       - name: Build unit tests
-        run: |
+        run:
           cd unit-tests/
           bash run_tests.sh
 
-      - name: Generate code coverage
-        run: |
-          cd unit-tests/
-          bash gen_coverage.sh
-
-      - uses: actions/upload-artifact@v2
-        with:
-          name: code-coverage
-          path: unit-tests/coverage
-
   job_test:
     name: Test
     needs: job_build_debug
+    strategy:
+      matrix:
+        device: [nanos, nanox, nanosp]
+        include:
+          - device: nanos
+            build_dir: nanos
+          - device: nanosp
+            build_dir: nanos2
+          - device: nanox
+            build_dir: nanox
```

Comment:
The include is only to change the `build_dir` of `nanosp` (nano S+) to `nanos2` since this is the internal name for the build directory.